plugins/catalog/src/alpha/entityContents.tsx

```diff
@@ -19,38 +19,85 @@ import {
-  EntityContentBlueprint.makeWithOverrides({
-    name: 'overview',
-    inputs: {
-      cards: createExtensionInput([
-        coreExtensionData.reactElement,
-        EntityContentBlueprint.dataRefs.filterFunction.optional(),
-        EntityContentBlueprint.dataRefs.filterExpression.optional(),
-      ]),
-    },
-    factory: (originalFactory, { inputs }) => {
-      return originalFactory({
-        defaultPath: '/',
-        defaultTitle: 'Overview',
-        loader: async () =>
-          import('./EntityOverviewPage').then(m => (
-            <m.EntityOverviewPage
-              cards={inputs.cards.map(c => ({
-                element: c.get(coreExtensionData.reactElement),
-                filterFunction: c.get(
+import {
+  EntityCardBlueprint,
+  EntityContentBlueprint,
+  OverviewEntityContentLauyoutBlueprint,
+} from '@backstage/plugin-catalog-react/alpha';
+import { buildFilterFn } from './filter/FilterWrapper';
+import { useEntity } from '@backstage/plugin-catalog-react';
+
+const catalogOverviewEntityContent = EntityContentBlueprint.makeWithOverrides({
+  name: 'overview',
+  inputs: {
+    layouts: createExtensionInput([
+      OverviewEntityContentLauyoutBlueprint.dataRefs.areas,
+      OverviewEntityContentLauyoutBlueprint.dataRefs.defaultArea,
+      OverviewEntityContentLauyoutBlueprint.dataRefs.filterFunction.optional(),
+      OverviewEntityContentLauyoutBlueprint.dataRefs.filterExpression.optional(),
+      OverviewEntityContentLauyoutBlueprint.dataRefs.component,
+    ]),
+    cards: createExtensionInput([
+      coreExtensionData.reactElement,
+      EntityContentBlueprint.dataRefs.filterFunction.optional(),
+      EntityContentBlueprint.dataRefs.filterExpression.optional(),
+      EntityCardBlueprint.dataRefs.area.optional(),
+    ]),
+  },
+  factory: (originalFactory, { inputs }) => {
+    return originalFactory({
+      defaultPath: '/',
+      defaultTitle: 'Overview',
+      loader: async () => {
+        const defaultLayout = await import('./EntityOverviewPage').then(
+          m => m.EntityOverviewPage,
+        );
+
+        const Component = () => {
+          const { entity } = useEntity();
+
+          // Use the first layout that matches the entity filter
+          const layout = inputs.layouts.find(l => {
+            const filterFunction = l.get(
+              OverviewEntityContentLauyoutBlueprint.dataRefs.filterFunction,
+            );
+            const filterExpression = l.get(
+              OverviewEntityContentLauyoutBlueprint.dataRefs.filterExpression,
+            );
+
+            return buildFilterFn(filterFunction, filterExpression)(entity);
+          });
+
+          const Layout =
+            layout?.get(
+              OverviewEntityContentLauyoutBlueprint.dataRefs.component,
+            ) ?? defaultLayout;
+
+          return (
+            <Layout
+              buildFilterFn={buildFilterFn}
+              cards={inputs.cards.map(card => ({
```

Comment:
I think it would be great if layout didn't have to check if the card should be rendered for a particular entity. Do you think the cards should be filtered before being passed to the layout component?

Reply:
Done